lms/djangoapps/django_comment_client/base/tests.py

```diff
@@ -968,6 +991,98 @@ def _test_unicode_data(self, text, mock_request):
         self.assertEqual(mock_request.call_args[1]["data"]["body"], text)
 
 
+@override_settings(MODULESTORE=TEST_DATA_MOCK_MODULESTORE)
+class ForumEventTestCase(ModuleStoreTestCase, MockRequestSetupMixin):
+    """
+    Forum actions are expected to launch analytics events. Test these here.
+    """
+    def setUp(self):
+        super(ForumEventTestCase, self).setUp()
+        self.course = CourseFactory.create()
+        seed_permissions_roles(self.course.id)
+        self.student = UserFactory.create()
+        CourseEnrollmentFactory(user=self.student, course_id=self.course.id)
+
+    @patch('eventtracking.tracker.emit')
+    @patch('lms.lib.comment_client.utils.requests.request')
+    def test_thread_event(self, __, mock_emit):
+        request = RequestFactory().post(
+            "dummy_url", {"thread_type": "discussion", "body": "Test text", "title": "Test"}
+        )
+        request.user = self.student
+        request.view_name = "create_thread"
+
+        views.create_thread(request, course_id=self.course.id.to_deprecated_string(), commentable_id="test_commentable")
+
+        event_name, event = mock_emit.call_args[0]
+        self.assertEqual(event_name, 'edx.forum.thread.created')
```

Comment:
ah, just read further down and this is an old question, though it doesn't look resolved.  not sure who's call it is to include that attribute and under what name, but it seems to me like it would be as interesting as anything else for research purposes.